File: review-service/src/main/java/onlineTravelBooking/review_service/entity/SupportTicket.java
package onlineTravelBooking.review_service.entity;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SupportTicket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long ticketId;

    private Long userId;

    @Column(length = 1000)
    private String issue;

    private String status;

    private Long assignedAgentId;

    private LocalDateTime createdAt;
}

File: frontend/online-travel-and-hospitality-booking-system/src/app/models/supportTicket.ts
export interface SupportTicket {
  ticketId: number;
  issue: string;
  status?: string;
  assignedAgentId?: number | null;
  userId?: number;
  createdAt?: string;
}
export type CreateSupportTicket = Omit<SupportTicket, 'ticketId'>;
